### PR TITLE
feat: [CDS-83866]: Create V1 Inputset merge open api

### PR DESCRIPTION
**Pull request by utkarshchoubey-harness**
feat: [CDS-83866]: Create V1 Inputset merge open api

Returns merged inputset response
```
{
    "inputs_yaml_merged": "env: qa\nenv2: test\ntimeout: 40m\n",
    "merged_pipeline_yaml": "version: 1\nkind: pipeline\nspec:\n  stages:\n    - type: custom\n      spec:\n        steps:\n          - type: plugin\n            spec:\n              uses: http\n              timeout: <+ inputs.timeout>\n              with:\n                url: http://<+ inputs.env>.harness.io\n                method: GET\n          - type: plugin\n            spec:\n              uses: http\n              timeout: <+ inputs.timeout>\n              with:\n                url: http://<+ inputs.env2>.harness.io/<+ inputs.env2>\n                method: GET\n                header: <+ input.env>\n  inputs:\n    env:\n      type: string\n      required: true\n      default: qa\n    env2:\n      type: string\n      required: true\n      default: prod\n    timeout:\n      type: string\n      required: true\n      default: 10m\n",
    "is_error_response": false,
    "inputset_error_wrapper": null
}
```

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

[CDS-83866]: https://harness.atlassian.net/browse/CDS-83866?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/55339)

### CODE REVIEW

**Comment by utkarshchoubey-harness** [comment]
trigger uts

**Review by archit-harness — CHANGES_REQUESTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55339)* status: 4 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 3047 at r2](https://reviewable.io/reviews/harness/harness-core/55339#-NjGjY266xwKBqrvgP06:-NjGjY266xwKBqrvgP07:bjbx2a3) ([raw file](https://github.com/harness/harness-core/blob/0155369637e06b074ab190cfab01d9480dcb15f4/pipeline-service/contracts/openapi/v1/openapi.yaml#L3047)):*
> ```YAML
>           x-stoplight:
>             id: p8ft8baj33lvh
>           description: Pipeline YAML after merging with the Input Sets
> ```

why we have 2 similar field?



_Code quote:_
```YAML
        pipeline_yaml:
          type: string
          x-stoplight:
            id: 8j3mw4nysjz5x
          description: Merged YAML of all the Input Sets
        complete_pipeline_yaml:
          type: string
          x-stoplight:
            id: p8ft8baj33lvh
          description: Pipeline YAML after merging with the Input Sets
```

**Comment by utkarshchoubey-harness** [comment]
trigger uts

**Review by archit-harness — DISMISSED** [review]
> Reviewed 1 of 3 files at r2, 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55339)* status: 4 file(s) to review. 1 of 2 approvals obtained. 

___
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 3047 at r2](https://reviewable.io/reviews/harness/harness-core/55339#-NjGjY266xwKBqrvgP06:-NjGtYOy0cZh5uKBV34e:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/0155369637e06b074ab190cfab01d9480dcb15f4/pipeline-service/contracts/openapi/v1/openapi.yaml#L3047)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

why we have 2 similar field?
</blockquote></details>

done

**Comment by utkarshchoubey-harness** [comment]
trigger uts

**Review by archit-harness — APPROVED** [review]
> Reviewed 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55339)* status: 4 file(s) to review. 1 of 2 approvals obtained.

File: pipeline-service/contracts/openapi/v1/openapi.yaml [hunk 1]
```diff
@@ -851,17 +851,56 @@ paths:
...
       parameters:
         - $ref: '#/components/parameters/AccountHeader'
       description: ''
       x-internal: true
+  '/v1/orgs/{org}/projects/{project}/input-sets/merge':
+    parameters:
+      - $ref: '#/components/parameters/OrgParam'
+      - $ref: '#/components/parameters/ProjectParam'
+    post:
+      summary: ''
```
Comment (mankrit-singh) [inline pipeline-service/contracts/openapi/v1/openapi.yaml:864]: please add summary

**Review by mankrit-singh — APPROVED** [review]

**Comment by utkarshchoubey-harness** [comment]
trigger review

**Comment by utkarshchoubey-harness** [comment]
trigger messagecheck

**Comment by utkarshchoubey-harness** [comment]
unrelated UTs are failing on the pr 
posted here - https://harness.slack.com/archives/C01JMN5P7EX/p1700040030442489

**Pull request merged**